src/diffusers/models/vq_diffusion_attention.py

```diff
@@ -0,0 +1,246 @@
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from diffusers.configuration_utils import ConfigMixin, register_to_config
+from diffusers.modeling_utils import ModelMixin
+
+from .attention import CrossAttention
+
+
+class VQDiffusionTransformer(ModelMixin, ConfigMixin):
```

Comment:
Could we try to re-use https://github.com/huggingface/diffusers/blob/888468dd90b4e353225d7d505fa28d2963e65678/src/diffusers/models/attention.py#L107 here as much as possible? #784